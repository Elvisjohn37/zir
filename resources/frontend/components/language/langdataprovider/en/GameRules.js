import React from 'react';
import NestedIntlProvider from 'frontend/components/NestedIntlProvider';

const MESSAGES = {
	game_rules: 'Gaming Rules',
	applicability: {
		title: 'Applicability of these Rules',
		description1:
			'The following product-specific rules (the "Gaming Rules") ' +
			'governthe end user\'s (the "Player", "you" or "your") ' +
			"use of all the Operator's (or SBOBET Casino's) " +
			'interactive casino products and services ' +
			'(the "Casino Games") available at ',
		description2: 'casino.sbobet.com',
		description3:
			' (the "Website"). They form part of the Operator\'s ' +
			'Terms and Conditions, which apply to all the Casino Games ' +
			'the Operator offers and which the Player has to accept ' +
			'upon registering for any of the Casino Games. ' +
			'To the extent there is any inconsistency between the ' +
			'Terms and Conditions and any of these Gaming Rules, ' +
			'the Terms and Conditions shall prevail.',
	},
	generalRules: 'GENERAL GAMING RULES',
	liability: {
		title: 'Liability',
		description:
			'The Player shall have no cause of action and ' +
			'hereby waives any rights or claims against the software developer ' +
			'of the Casino Games for any matter, cause or thing involving the ' +
			'Player’s participation in the Casino Games or otherwise.',
	},
	playForFun: {
		title: ' Play for fun',
		description:
			'The Player agrees that the Casino Games are for ' +
			'entertainment value only. The Player understands and ' +
			'acknowledges that no monetary bet is necessary or required ' +
			'to play the Casino Games. If the Player wishes to play without ' +
			'betting money, they may do so in the "demo play" area only.',
	},
	personalUse: {
		title: ' Personal use only',
		description:
			"The Player's interest in the Casino and the Website " +
			'is personal and not professional. A Player entering the Website ' +
			'does so solely for their own personal entertainment and any other ' +
			'entrance, access, use or re-use of theCasino Games is strictly ' +
			'prohibited.',
	},
	malfunctions: {
		title: 'Malfunction',
		description:
			'Unless otherwise specified, below (e.g. with certain ' +
			'live games) malfunctions of any sort (software or hardware) ' +
			'will void play. This means that any stake placed will be returned ' +
			'irrespective of any indicative result.',
	},
	smartPlayer: {
		title: 'Smart Player and Artificial Intelligence',
		description:
			'SBOBET Casino reserves the right to reject SMART ' +
			'players or any suspected SMART players and Card Counters. ' +
			'Any activities using artificial intelligencecalled bots are ' +
			'strictly prohibited. Any attempt to breach or violate this policy ' +
			"will result in suspension and exclusion of the Player's Account. " +
			'All winnings and commissions will be forfeited.',
	},
	complaintsRng: {
		title: 'Complaint for random number generated gaming',
		description1:
			'If a Player has any complaints about any aspect of ' +
			'the Casino Games (save for the Live Casino Games, the complaints ' +
			'procedure for which is dealt with below), they should submit the ' +
			'nature such complaint within 14 (fourteen) days of the incident ' +
			'occurring to ',
		description2: 'support@sbobet.com',
		description3:
			' together with their User ID, the ' +
			'time and date of playing and any further information that may be ' +
			'relevant. Please note that the Operator will investigate the ' +
			'complaint fully using data from the Operator’s servers but the ' +
			'ultimate decision on the complaint rests with the Operator acting ' +
			'reasonably in light of all the available evidence.',
	},
	complaintsLg: {
		title: 'Complaints for Live Gaming',
		description1:
			'If a Player wishes to make a complaint or dispute a ' +
			'live Casino Game result, they must provide the Operator with their ' +
			"user ID the time of playing, the dealer's name, the Table ID and " +
			'Round ID at the time of contacting the Operator.Failure to do so ' +
			'may result in the complaint being unable to be addressed by the ' +
			'Operator.Video images of live Casino Games is kept for 24 hours ' +
			'and therefore Players must address their complaint within 24 hours ' +
			'of the dispute occurring. Any complaints submitted after 24 hours ' +
			'will be rejected by the Operator due to the absence of video evidence',
		description2:
			'In case of any dispute, the Player acknowledges and agrees that ' +
			'the Operator’s decision is final and official.',
	},
	maxWin: {
		title: 'Maximum Win',
		description: 'There is no maximum gross win in any day.',
	},
	liveCasinoGames: {
		title: 'Live Casino Games',
		description1:
			'For Live Casino Games, valid results on the Casino ' +
			'Games are those results which are detected by the electronic ' +
			'sensor equipment installed for that purpose.If for any reason ' +
			'a result is not detected and registered by the electronic sensors, ' +
			'then that result is deemed to have not occurred, and any bets ' +
			'locked will remain locked until a valid result is determined.',
		description2:
			'Specific rules on any malfunction in Live Casino ' +
			'Games are included in the specific Casino Games rules below. ' +
			'Please left-click on the names of any of the Casino Games to ' +
			'see the specific rules for that Casino Game:',
	},
	casinoGames: {
		live_dealer_baccarat: 'LIVE DEALER BACCARAT',
		live_dealer_roulette: 'LIVE DEALER ROULETTE',
		live_dealer_sicbo: 'LIVE DEALER SIC BO',
		live_dealer_blackjack: 'LIVE DEALER BLACKJACK',
		games: 'GAMES',
		slot: 'SLOT',
		card: 'CARD',
		table: 'TABLE',
	},
};

export default function Provider({ children, ...props }) {
	return (
		<NestedIntlProvider {...props} messages={MESSAGES}>
			{children}
		</NestedIntlProvider>
	);
}
