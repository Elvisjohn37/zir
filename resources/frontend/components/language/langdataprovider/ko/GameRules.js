import React from 'react';
import NestedIntlProvider from 'frontend/components/NestedIntlProvider';

const MESSAGES = {
	game_rules: '게임 규칙',
	applicability: {
		title: '본 규칙의 적용 범위',
		description1: '다음의 게임별 규칙("게임 규칙")은 ',
		description2: 'casino.SBOBET.com ',
		description3:
			'("웹사이트")에서 사용이 가능한 최종사용자("플레이어", ' +
			'플레이어의")의 운영자의(또는 SBOBET 카지노의) 모든 인터액티브 카지노 ' +
			'상품 및 서비스("카지노 게임")에 적용됩니다. 이것은 운영자가 제공하는 ' +
			'모든 카지노 게임에 적용되며 플레이어가 카지노 게임에 등록 시 동의해야 ' +
			'하는 운영자의 사용조건에 적용됩니다. 사용조건과 게임 규칙 간에 상충되는 ' +
			'내용이 있는 경우, 사용조건이 우선합니다.',
	},
	general_rules: '일반적 게임 규칙',
	liability: {
		title: '책임',
		description:
			'플레이어는 소송의 사유를 가질 수 없으며 이에 플레이어는 ' +
			'본인의 카지노 게임에 대한 참여 및 기타와 관련한 어떠한 사항, 사유나 ' +
			'사물에 대해 카지노 게임의 소프트웨어 개발자에 대한 권한이나 청구를 포기합니다.',
	},
	playForFun: {
		title: '재미를 위한 플레이',
		description:
			'플레이어는 카지노 게임은 단순한 재미의 가치만 가지고 있다는 ' +
			'것에 동의합니다. 플레이어는 카지노 게임를 플레이하기 위해 금전적 베팅이 ' +
			'필요하지 않다는 것을 이해 및 인정합니다. 플레이어가 돈을 걸지 않고 ' +
			'플레이하고자 하는 경우, 이들은 "데모 플레이"에서만 플레이할 수 있습니다.',
	},
	personalUse: {
		title: '개인 사용 전용',
		description:
			'카지노와 웹사이트에 대한 플레이어의 관심은 개인적인 것이며 ' +
			'전문성을 띠지 않습니다. 웹사이트에 입장하는 플레이어는 개인적인 재미를 ' +
			'위한 것이며 그 외의 목적으로 카지노 게임에 입장, 접속, 사용 또는 ' +
			'재사용하는 것은 엄격하게 금지됩니다.',
	},
	malfunctions: {
		title: '오작동',
		description:
			'달리 명시되지 않는 한 아래의(예: 특정 라이브 게임에서와 ' +
			'같이) 어떠한 종류(소프트웨어나 하드웨어)의 오작동은 플레이를 무효로 ' +
			'합니다. 이는 걸었던 모든 내기금은 결과가 어떻게 될 것처럼 보였는지 와는 ' +
			'관계 없이 전부 돌려준다는 의미입니다..',
	},
	smartPlayer: {
		title: '부정선수및인공지능플러그',
		description:
			'SBOBET 카지노는플레이어의베팅을속임수또는부정플레이어와테' +
			'이블게임을모니터링을거부할권리를보유합니다. SBOBET 카지노는부정선수및인공지능플러' +
			'그행위의사용은위반및선수들의규정위반시도, 계정이비활성화되고삭제됩니다허용하지않습' +
			'니다. 자신의위닝은잃게될것이다.',
	},
	complaintsRng: {
		title: '난수 발생 게임에 대한 불만',
		description1:
			'플레이어가 카지노 게임의 어떠한 부분에 대해 불만이 있는 경우(라이브 ' +
			'카지노 게임 제외함. 이에 대한 불만 제기 절차는 아래에서 명시함), 해당 불만이 ' +
			'발생한 후 14(십사)일 이내에 이의 성격을 사용자 ID, 플레이 날짜와 시간, 관련된 그 밖의 정보와 함께',
		description2: 'support@sbobet.com',
		description3:
			'에 제출해야 합니다. 운영자는 운영자 서버의 데이터를 충분히 활용하여 해당 ' +
			'불만 사항을 조사할 것이지만 불만에 대한 결정은 모든 입수 가능한 정보를 기반으로 합리적인 ' +
			'행동을 취하는 운영자가 하게 됩니다.',
	},
	complaintsLg: {
		title: '라이브 게임 불만',
		description1:
			'플레이어가 라이브 카지노 게임 결과에 대해 불만이나 반대를 제기하고자 하는 경우, ' +
			'플레이어는 운영자에게 연락 시 자신의 사용자 ID, 플레이 날짜와 시각, 딜러 이름, 테이블 ID 및 ' +
			'라운드 ID를 운영자에게 제공해야 하며 그러지 못할 경우 운영자가 해당 불만을 해결하지 못할 수도 ' +
			'있습니다. 라이브 카지노 게임에 대한 비디오 이미지는 24시간 동안 보관하므로 플레이어는 해당 ' +
			'분쟁 발생 시 24시간 이내에 불만을 제기해야 합니다. 24시간 후에 제기되는 일체의 불만은 비디오 ' +
			'증거 자료가 부재하므로 운영자에 의해 거절됩니다.',
		description2:
			'분쟁이 발생한 경우, 플레이어는 운영자의 결정이 최종적이며 공식적인 것임을 인정하고 그에 동의합니다.',
	},
	maxWin: {
		title: '최대 상금',
		description: '1일 최대 상금 한도는 없습니다.',
	},
	liveCasinoGames: {
		title: '라이브 카지노 게임',
		description1:
			'라이브 카지노 게임에서, 유효한 카지노 게임의 결과란 해당 목표로 설치된 전자 ' +
			'센서 장비에 의해 감지되는 결과를 말합니다. 어떠한 이유에서건 전자 센서에 의해 결과가 감지 ' +
			'및 등록이 되지 않는 경우 해당 결과는 발생하지 않은 것으로 간주하며 걸어둔 일체의 베팅금은 ' +
			'유효한 결과가 결정될 때까지 걸어둔 상태를 유지합니다.',
		description2:
			'라이브 카지노 게임의 오작동에 대한 상세 규칙은 아래의 상세 카지노 게임 ' +
			'규칙에 명시되어 있습니다. 해당 카지노 게임의 상세 규칙을 보려면 해당 게임의 이름에 대고 ' +
			'왼쪽 마우스 버튼을 클릭하면 됩니다.:',
	},
	casinoGames: {
		live_dealer_baccarat: '라이브 딜러 바카라',
		live_dealer_roulette: '라이브 딜러 룰렛',
		live_dealer_sicbo: '라이브 딜러 식보',
		live_dealer_blackjack: '라이브 딜러 블랙잭',
		games: 'GAMES',
		slot: 'SLOT',
		card: 'CARD',
		table: 'TABLE',
	},
};

export default function Provider({ children, ...props }) {
	return (
		<NestedIntlProvider {...props} messages={MESSAGES}>
			{children}
		</NestedIntlProvider>
	);
}
