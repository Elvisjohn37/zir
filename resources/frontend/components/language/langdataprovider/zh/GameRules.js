import React from 'react';
import NestedIntlProvider from 'frontend/components/NestedIntlProvider';

const MESSAGES = {
	game_rules: '游戏规则',
	applicability: {
		title: '本规则适用范围',
		description1:
			'以下游戏的具体规则（以下简称“游戏规则” ）适用于最终用户（以下简称“玩家” ， ' +
			'“你”或“你们” ）对所有运营商（或 SBOBET 网上娱乐场）交互式赌场的产品和服务（以下简称“娱乐场游戏“ ）',
		description2: 'http://casino.sbobet.com ',
		description3:
			'（以下简称”网站“ ）的使用。这些游戏规则是运营商附带条款的一部分。这些附带条款适用于' +
			'运营商提供的所有娱乐场游戏，玩家在任何娱乐场游戏注册时，都必须接受这些条款。如果游戏规则和附带条款有' +
			'任何不一致的地方，以附带条款为准。',
	},
	general_rules: '一般游戏规则',
	liability: {
		title: '责任和义务',
		description: '不论玩家在玩游戏的过程中出现任何情况，玩家都无权对娱乐场的开发商有任何不利的行为。',
	},
	playForFun: {
		title: '为娱乐而游戏',
		description:
			'玩家必须同意，娱乐场游戏仅供娱乐。玩家应理解并认识到，参加娱乐场游戏不一定必须赌钱。' +
			'如果玩家仅想游戏不想赌钱，他们可以在“演示播放”区域内进行。',
	},
	personalUse: {
		title: '仅供个人使用',
		description: '玩家对娱乐场及其网站的兴趣是基于个人而非基于某种专业目的。玩家应仅仅是出',
	},
	malfunctions: {
		title: '故障',
		description:
			'除非另有说明，以下（如某些真人游戏）的任何故障（软件或硬件）将导致游戏无效。' +
			'这意味着不论有任何结果，任何赌注都将返还。',
	},
	smartPlayer: {
		title: '作弊玩家和人工智能插件',
		description:
			'SBOBET 赌场将保留权利拒绝作弊玩家的投注或监视作弊玩家和赌台。 ' +
			'SBOBET 赌场不容许任何作弊玩家和使用人工智能插件的行径， 任何尝试违反和侵犯该条例的玩家，' +
			'账户将会被停用和删除。其彩金也会被没收。',
	},
	complaintsRng: {
		title: '对随机生产数字游戏的投诉',
		description1:
			'如果玩家对娱乐场游戏有任何方面的投诉（现场娱乐场游戏除外，其投诉程序如下） ，' + '他们应该在事发14天内向 ',
		description2: 'support@sbobet.com ',
		description3:
			' 提交投诉，提交的投诉应包括其用户名，' +
			'游戏的日期和时间及任何进一步的有关资料。请注意，经营者将会通过服务器上的数据全面调查这项投诉，' +
			'但对投诉的最终决定在于经营者合理地考虑到所有现有的证据。',
	},
	complaintsLg: {
		title: '对现场游戏的投诉',
		description1:
			'如果玩家想对现场娱乐场游戏结果提出申诉或质疑，他们必须向经营者提供其用户名，' +
			'游戏的时间，荷官的名字，赌桌名及出问题的游戏代号，否则经营者无法处理投诉。' +
			'现场娱乐场游戏的视频图像会保存24小时，因此，玩家必须在争执发生的24小时内进行投诉。' +
			'任何24小时后的投诉将会因为没有录像证据而被经营者拒绝。',
		description2: '如有任何争议，玩家确认并同意，经营者的决定是最终并且正式的决定。',
	},
	maxWin: {
		title: '赢利的最高限额',
		description: '任何一天都没有赢利的最高限额',
	},
	liveCasinoGames: {
		title: '现场娱乐场游戏',
		description1:
			'对于现场娱乐场游戏来说，只有那些被所安装的电子传感器设备所探测到的结果才是游戏的游戏结果。' +
			'如果由于某种原因，游戏结果没有被电子传感器检测到或记录到，那么这个结果就被认定是没有发生过，' +
			'而赌注则会被锁死，直到有效结果出现。',
		description2:
			'现场娱乐场的故障规则包括在以下具体的游戏规则中。' +
			'请鼠标左键点击任何游戏的名称来查看各个游戏的特定规则：',
	},
	casinoGames: {
		live_dealer_baccarat: '现场百家乐',
		live_dealer_roulette: '现场轮盘',
		live_dealer_sicbo: '现场骰宝',
		live_dealer_blackjack: '在线庄家 21 点',
		games: 'GAMES',
		slot: 'SLOT',
		card: 'CARD',
		table: 'TABLE',
	},
};

export default function Provider({ children, ...props }) {
	return (
		<NestedIntlProvider {...props} messages={MESSAGES}>
			{children}
		</NestedIntlProvider>
	);
}
