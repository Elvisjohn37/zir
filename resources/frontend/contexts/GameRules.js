import React, { useEffect, useState } from 'react';
import gameRulesReducer from './gamerules/gameRulesReducer';
import { LangConsumer } from 'frontend/contexts/Language';
import { getGameGuide } from 'frontend/ajax/backend';

let GameRulesContext = React.createContext();

export function GameRulesProviderRaw({ children, lang }) {
	let [gameRulesState, gameRulesDispatch] = React.useReducer(gameRulesReducer, {
		isLoading: true,
		gameRulesList: [],
		isError: false,
		config: {},
	});
	let [isMounted, setIsMounted] = useState(false);

	useEffect(() => {
		if (!isMounted) {
			setIsMounted(true);
			getGameGuide({
				isErrorHandled: true,
				lang: lang,
				success: (response) => {
					gameRulesDispatch({
						type: 'SET_LIST',
						data: response.data.data,
					});
				},
				error: () => {
					gameRulesDispatch({
						type: 'SET_ISERROR',
						isError: true,
					});
				},
			});
		}
	}, [isMounted]);

	return (
		<GameRulesContext.Provider
			value={{
				gameRulesState,
				gameRulesDispatch,
			}}
		>
			{children}
		</GameRulesContext.Provider>
	);
}

export default function GameRulesProvider({ children }) {
	return (
		<LangConsumer>
			{({ langState }) => <GameRulesProviderRaw lang={langState.active}>{children}</GameRulesProviderRaw>}
		</LangConsumer>
	);
}

export let GameRulesConsumer = GameRulesContext.Consumer;
