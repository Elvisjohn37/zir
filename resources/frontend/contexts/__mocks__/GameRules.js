import * as gameRulesHelper from 'frontend/tests/gameRulesHelper';
let mockedModule = jest.createMockFromModule('../GameRules.js');

mockedModule.gameRulesDispatch = () => {};

// main component
mockedModule.GameRulesConsumer = ({ children }) => {
	return children({
		gameRulesState: gameRulesHelper.getGameRulesState(),
		gameRulesDispatch: mockedModule.gameRulesDispatch,
	});
};
mockedModule = { ...mockedModule, ...gameRulesHelper };

module.exports = mockedModule;
