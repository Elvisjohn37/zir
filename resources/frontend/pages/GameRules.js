import React from 'react';
import Link from '@mui/material/Link';
import { NavLink } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { LangDataProvider } from 'frontend/components/Language';
import styles from './GameRules.module';
import { GameRulesConsumer } from 'frontend/contexts/GameRules';
import GameRulesLoader from 'frontend/components/GameRulesLoader';
import lodash from 'lodash';
import { extractProductGameType } from './gamerules/helper';
import GameRulesErrorMessage from 'frontend/components/GameRulesErrorMessage';

export default function GameRules() {
	return (
		<div className={styles.gameRules}>
			<GameRulesConsumer>
				{({ gameRulesState }) =>
					gameRulesState.isLoading ? (
						<GameRulesLoader />
					) : gameRulesState.isError ? (
						<GameRulesErrorMessage />
					) : (
						<LangDataProvider category="gamerules" className="testGamingRules">
							<div className={styles.gamingRules}>
								<h3>
									<FormattedMessage id="applicability.title" />
								</h3>
								<div className={styles.content}>
									<FormattedMessage id="applicability.description1" />
									<Link href="/" className="testHomePage">
										<FormattedMessage id="applicability.description2" />
									</Link>
									<FormattedMessage id="applicability.description3" />
								</div>
								<h3>
									<FormattedMessage id="generalRules" />
								</h3>
								<ol>
									<h3>
										<li>
											<FormattedMessage id="liability.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="liability.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="playForFun.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="playForFun.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="personalUse.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="personalUse.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="malfunctions.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="malfunctions.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="smartPlayer.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="smartPlayer.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="complaintsRng.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="complaintsRng.description1" />
										<Link href="mailto:support@sbobet.com">
											<FormattedMessage id="complaintsRng.description2" />
										</Link>
										<FormattedMessage id="complaintsRng.description3" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="complaintsLg.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="complaintsLg.description1" />
										<br />
										<br />
										<FormattedMessage id="complaintsLg.description2" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="maxWin.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="maxWin.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="liveCasinoGames.title" />
										</li>
									</h3>

									<div className={styles.content}>
										<FormattedMessage id="liveCasinoGames.description1" />
										<br />
										<br />
										<FormattedMessage id="liveCasinoGames.description2" />
										<br />
										<br />

										<ol>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#BACCARAT'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_baccarat" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#ROULETTE'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_roulette" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#SICBO'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_sicbo" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#BLACKJACK'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_blackjack" />
												</Link>
											</li>
											{extractProductGameType(gameRulesState.gameRulesList).map((item, index) => (
												<li key={index} className={styles.gameRulesProduct}>
													<FormattedMessage
														id={`casinoGames.${lodash.camelCase(item.productName)}`}
														defaultMessage={item.productName.toUpperCase()}
													/>
													<ol>
														{item.gameTypes.map((gameTypeItem, gameTypeIndex) => (
															<li key={`gameType${gameTypeIndex}`} className={styles.gameRulesGameType}>
																<Link
																	component={NavLink}
																	to={`/gamerules/${lodash.kebabCase(item.productName)}/${lodash.kebabCase(
																		gameTypeItem
																	)}`}
																>
																	<FormattedMessage
																		id={`casinoGames.${lodash.camelCase(gameTypeItem)}`}
																		defaultMessage={gameTypeItem}
																	/>
																</Link>
															</li>
														))}
													</ol>
												</li>
											))}
										</ol>
									</div>
								</ol>
							</div>
						</LangDataProvider>
					)
				}
			</GameRulesConsumer>
		</div>
	);
}
