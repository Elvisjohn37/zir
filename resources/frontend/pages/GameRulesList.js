import React from 'react';
import { GameRulesConsumer } from 'frontend/contexts/GameRules';
import { LangDataProvider } from 'frontend/components/Language';
import GameRulesListRaw from './gameruleslist/GameRulesListRaw';

export default function GameRulesList() {
	return (
		<LangDataProvider category="gamerulescontent">
			<GameRulesConsumer>
				{({ gameRulesState }) => <GameRulesListRaw gameRulesState={gameRulesState} />}
			</GameRulesConsumer>
		</LangDataProvider>
	);
}
