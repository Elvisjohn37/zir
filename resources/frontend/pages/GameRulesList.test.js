import React from 'react';
import GameRulesList from './GameRulesList';
import { render } from '@testing-library/react';
import { getGameRulesState } from 'frontend/contexts/GameRules';

jest.mock('frontend/contexts/GameRules');
jest.mock('frontend/components/Language');
jest.mock('./gameruleslist/GameRulesListRaw');

describe('<GameRulesList />', () => {
	it('should match snapshot', () => {
		let { asFragment } = render(<GameRulesList />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should display game rules list with correct data', () => {
		let gameRulesState = getGameRulesState();
		let { getByTestId } = render(<GameRulesList />);
		let gameRulesList = getByTestId('GameRulesListRaw');

		expect(gameRulesList).toHaveAttribute('gamerulesstate', JSON.stringify(gameRulesState));
	});
});
