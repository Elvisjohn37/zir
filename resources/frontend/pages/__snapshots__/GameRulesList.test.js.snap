// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameRulesList /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="gamerulescontent"
    data-testid="LangDataProvider"
  >
    <div>
      gamerulescontent
    </div>
    <div
      data-testid="GameRulesListRaw"
      gamerulesstate="{\\"config\\":{\\"liveDealerGameLink\\":\\"http://info.sbobet.com/article/AA-00363/41/\\"},\\"gameRulesList\\":[{\\"gameName\\":\\"Dice Wars (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"Djap Go (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":1},{\\"gameName\\":\\"Sedie (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"Three Stars Blessing (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":1},{\\"gameName\\":\\"Whack d' Mole (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"Three Card Poker (Mobile)\\",\\"productName\\":\\"Multiplayer\\",\\"gameTypeName\\":\\"Card Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"Dragon Tiger (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Card Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"3 Faces Baccarat (Mobile)\\",\\"productName\\":\\"Multiplayer\\",\\"gameTypeName\\":\\"Card Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":1},{\\"gameName\\":\\"Mystic Cards (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Card Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0},{\\"gameName\\":\\"Yin Yang Treasure (Mobile)\\",\\"productName\\":\\"Games\\",\\"gameTypeName\\":\\"Table Games\\",\\"url\\":\\"https://docs.google.com/document/d/e/2PACX-1vRV0Y-UzPEwdj_2ebm7a_Laiq-B7C4PbIMCjNYYdFPsZOWSc-cofIdL_T49p13_sAVHlAJJmh1slDxM/pub?embedded=true\\",\\"isNew\\":0}],\\"isError\\":false,\\"isLoading\\":false}"
    />
  </div>
</DocumentFragment>
`;
